### PR TITLE
ARROW-15936 [Ruby] Add test for Arrow::DictionaryArray#raw_records

### PR DESCRIPTION
**Pull request by okadakk**
I added test for Arrow::DictionaryArray#raw_records.

### CODE REVIEW

File: ruby/red-arrow/test/raw-records/test-dictionary-array.rb [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        value_data_type: dictionary.value_data_type,
+        ordered: false,
+      },
+    })
+  end
+
+  def build_record_batch(dictionary, records)
+    Arrow::RecordBatch.new(build_schema(dictionary),
+                           records.size,
+                          [dictionary.dictionary_encode])
```
Comment (kou) [inline ruby/red-arrow/test/raw-records/test-dictionary-array.rb:33]:
```suggestion
                           [dictionary.dictionary_encode])
```

File: ruby/red-arrow/test/raw-records/test-dictionary-array.rb [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+      column: {
+        type: :dictionary,
+        index_data_type: :int32,
+        value_data_type: dictionary.value_data_type,
+        ordered: false,
+      },
+    })
+  end
+
+  def build_record_batch(dictionary, records)
```
Comment (kou) [inline ruby/red-arrow/test/raw-records/test-dictionary-array.rb:30]: Umm. This `dictionary` isn't a dictionary.

For example, `Arrow::NullArray.new(records.flatten)` in `build(Arrow::NullArray.new(records.flatten), records)` isn't a dictionary.

How about something like the following?

```ruby
def build_record_batch(array)
  dictionary = array.dictionary_encode
  schema = Arrow::Schema.new(column: dictionary.value_data_type)
  Arrow::RecordBatch.new(schema, array.length, [dictionary])
end
```

```ruby
target = build(Arrow::NullArray.new(records.collect(&:first)))
```

**Review by kou — CHANGES_REQUESTED** [review]

**Review by kou — APPROVED** [review]
+1

**Comment by okadakk** [comment]
thanks! Sorry for the late reaction...

**Comment by kou** [comment]
No problem! But please mention me when a pull request can be reviewed again.

**Pull request closed without merging**